.github/workflows/check-traceability.yml

```
@@ -0,0 +1,47 @@
+name: Check PR traceability
+
+on:
+  push:
+    branches: [master, "release/*"]
+  pull_request:
+    branches: [master, "release/*"]
+
+jobs:
+  check-pr-description:
+    name: Check PR description
+    runs-on: ubuntu-20.04
+    steps:
+
+    - uses: actions/checkout@v2
+
+    - name: Set up Python 3.6
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.6
+
+    - name: Fetch PR Description
+      env:
+        GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+      run: echo "::set-output name=description::$(gh pr view ${{ github.event.pull_request.number }} | tr -d "\n\r" | tr -d ' ')"
+      id: pr
+
+    - name: Check PR validity
+      env:
+        PR_DESCRIPTION: ${{ steps.pr.outputs.description }}
+      shell: python
+      run: |
+        import os
+
+        description = os.getenv("PR_DESCRIPTION").lower()
+
+        if '<issue_number>' in description:
+          raise Exception("Please replace <issue_number> with a reference issue to track details of what this PR does!")
+
+        GITHUB_SPECIAL_LINKING_WORDS = ('close', 'closes', 'closed', 'fix', 'fixes', 'fixed', 'resolve', 'resolves', 'resolved')
+
+        contains_fix = any(f"{w}#" in description for w in GITHUB_SPECIAL_LINKING_WORDS)
+        contains_parts = 'partsof' in description
+        contains_follow_up = 'followupof' in description
+
+        if not (contains_fix or contains_parts or contains_follow_up):
+          raise Exception("Please link to an issue in the PR description to track details of what this PR!")
```

```suggestion
          raise Exception("Please link to an issue in the PR description to track details of what this PR does!")
```
the sentence is incomplete

---

isn't the exclamation mark a bit too aggressive?